threads-csharp/PeriodicTimerCSharp/PeriodicTimerCSharp/PeriodicTimerMethods.cs

```diff
@@ -0,0 +1,70 @@
+﻿namespace PeriodicTimerCSharp;
+
+public class PeriodicTimerMethods
+{
+    private readonly PeriodicTimer _periodicTimer;
+    private readonly CancellationTokenSource _cancellationToken = new();
+    private Task? _task;
+    private int _size;
+
+    public PeriodicTimerMethods(TimeSpan timeSpan, int size) 
+    {
+        _periodicTimer = new PeriodicTimer(timeSpan);
+        _size = size;
+    }
+
+    public static int[] CreateRandomArray(int size)
+    {
+        var array = new int[size];
+
+        for (int i = 0; i < size; i++)
+        {
+            array[i] = Random.Shared.Next();
+        }
+        
+        return array;
+    }
+
+    public void Start() 
+    {
+        _task = RandomArrayTimerAsync();
```

Comment:
Shouldn't this be awaited? Since the `RandomArrayTimerAsync()` method is asynchronous,

Reply:
Hi @mahrncic, we are invoking the Start method synchronously, since we assume that it's the first method we execute and we only use it to initialize our _task variable. On the other hand, the Stop method should be awaited to give time for any pending tasks to be completed. Let me know if it makes sense.